eks/modules/eks/worker-iam.tf

```diff
@@ -0,0 +1,36 @@
+
+data "aws_iam_policy_document" "worker-role-policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+  }
+}
+resource "aws_iam_role" "aws-node" {
+  name = "terraform-eks-aws-node"
```

Comment:
`name = "${var.cluster-name}-node"`
needs to be set through variable interpolation otherwise, the deployment will be only possible for one cluster setup.